packages/Parallel/Command/WorkerCommandLineFactory.php

```diff
@@ -61,6 +61,11 @@ public function create(
                 break;
             }
 
+            if ($arg === $mainScript) {
+                $workerCommandArray[] = '"' . $arg . '"';
+                continue;
+            }
```

Comment:
@paulbalandan I tried still, still doesn't work

```
➜  spaced prj sample git:(main) ✗ cd target
➜  target git:(main) ✗ ls
composer.json  composer.lock  rector.php  src  vendor
➜  target git:(main) ✗ vendor/bin/rector 
 0/1 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%{"fatal_errors":["The file \"/home/abdul/Downloads/spaced\" does not exist."]}
                                                                                
 [ERROR] Could not process some files, due to:                                  
         "Child process error: ".                                               
```

Feel free to provide a patch if you found the solution first :)

Reply:
What is the drawback of using relative path for main script? Can you give example?

Reply:
Using relative path will not work for global  usage, see example e2e on rector/rector scoped vendor

https://github.com/rectorphp/rector/blob/ca844057bcb6c096d4380a4a1624914a90b7de4b/.github/workflows/e2e_global.yaml#L39

Reply:
Also, using relative path make impossible to know the current path that going to be scanned since the parallel is different process

Reply:
I wonder wheter we need escapeshellcmd on the mainscript instead of escapeshellarg?

https://3v4l.org/XJILI

Reply:
the root path before project path is probably the one that need `escapeshellcmd()`, I will retry tomorrow